proposals/NNNN-submenu-icon.md

```diff
@@ -0,0 +1,121 @@
+   :
+   <param name="menuIcon" type="Common.Image" mandatory="false" />
+</function>
+```
+
+## Potential downsides
+
+The proposed change is backward compatible and will not cause a breaking change. However the API of `AddSubMenu` and `AddCommand` are inconsistent. The alternative considered is showing a consistent API which uses `MenuParams` but it's not fully backward compatible.
+
+## Impact on existing code
+
+The impact on existing code is very small.
+- an HMI implemntation can use the logic of a menu command to show a menu icon
+- Existing apps are not affected as this proposal is only adding a single parameter
+- Changes on SDL core are minimal
+- Only requires a minor version chagne 
+
+## Alternatives considered
+
+The alternative proposed solution is to use `MenuParams` to provide an image parameter to `AddCommand` and `AddSubMenu`. This could cause a breaking change. However the result would be a clean API.
+
+- For the struct `MenuParams` add a parameter called `menuIcon` of type `Image`
+- For the function `AddCommand` deprecate the parameter `cmdIcon`
+- For the function `AddSubMenu` add a parameter called `menuParams` of type `MenuParams` 
+- For the function `AddSubMenu` deprecate the parameters `position` and `menuName`
+
+The result would be a consistent API through both functions affecting the apps menu.
+
+### Mobile API
+
+```xml
+<struct name="MenuParams">
+  :
+  <param name="menuIcon" type="Image" mandatory="false">
+    <description>A new optional parameter to specify an icon to be shown in the menu entry</description>
+  </param>
+</struct>
+
+<function name="AddCommand" functionID="AddCommandID" messagetype="request">
+  :
+  <param name="cmdIcon" type="Image" mandatory="false">
+    <description>Deprecated parameter. Please use .menuParams.menuIcon instead.</description>
+  </param>
+</function>
+
+<function name="AddSubMenu" functionID="AddSubMenuID" messagetype="request">
+  :
+  <param name="menuParams" type="MenuParams">
+    <description>A new parameter containing the data of the menu entry.</description>
+  </param>
+  <param name="position" type="Integer" minvalue="0" maxvalue="1000" defvalue="1000" mandatory="false">
+    <description>Deprecated parameter. Please use .menuParams.position instead.</description>
+  </param>
+  <param name="menuName" maxlength="500" type="String">
+    <description>Deprecated parameter. Please use .menuParams.menuName instead.</description>
+  </param>
+</function>
+```
+
+### HMI API
+
+```xml
+<struct name="MenuParams">
+  :
+  <param name="menuIcon" type="Common.Image" mandatory="false">
+    <description>A new optional parameter to specify an icon to be shown in the menu entry</description>
+  </param>
+</struct>
+
+<function name="AddCommand" messagetype="request">
+  :
+  <param name="cmdIcon" type="Common.Image" mandatory="false">
+    <description>Deprecated</description>
+  </param>
+</function>
+```
+
+### Backward compatibility
```

Comment:
Does "Backward compatibility" only apply to the alternative solution, or would this be applied to the proposed solution?  If it applies to the proposed solution, please update the header to ## so that it's not in the "Alternatives considered" section of the proposal.  If it only applies to the alternative solution, this is fine as is.